src/Installers/Windows/Common/dotnethome_x64.wxs

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+  <?ifndef Platform?>
+    <?define Platform = "$(sys.BUILDARCH)"?>
+  <?endif?>
+
+  <!-- InstallerArchitecture matches the expected values for PROCESSOR_ARCHITECTURE
+       https://docs.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details  -->
+  <?if $(var.Platform)~=x86?>
+    <?define InstallerArchitecture="X86"?>
+  <?elseif $(var.Platform)~=x64?>
+    <?define InstallerArchitecture="AMD64"?>
+  <?elseif $(var.Platform)~=arm64?>
+    <?define InstallerArchitecture="ARM64"?>
+  <?else?>
+    <?error Unknown platform, $(var.Platform) ?>?
+  <?endif?>
+
+  <Fragment>
+    <!-- Identify when installing in emulation as when PROCESSOR_ARCHITECTURE does not match the installer architecture
+          https://docs.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details -->
+    <SetProperty Action="Set_NON_NATIVE_ARCHITECTURE" Id="NON_NATIVE_ARCHITECTURE" Value="true" Before="CostFinalize">
+      NOT %PROCESSOR_ARCHITECTURE="$(var.InstallerArchitecture)"
+    </SetProperty>
+  </Fragment>
+
+  <?if $(var.Platform)~=x64?>
+  <Fragment>
+    <!-- When running in a non-native architecture and user hasn't specified install directory,
+         install to an x64 subdirectory.
+         This is only define for x64, since x86 has redirection and no other native architecture can install ARM64 today -->
```

Comment:
Could you clarify "no other native architecture can install ARM64 today" a bit❔ Is there just one `Program Files` folder on ARM64 machines❔

Reply:
> Could you clarify "no other native architecture can install ARM64 today" a bit

So there are 3 architectures we support: x86, x64, and ARM64.  X86 does not need any redirection, since the OS does this for us.  Arm64 doesn't need any redirection since there is no other architecture that can install an ARM64 MSI (nor can we predict if one were created how the OS would treat it).  X64 is the only one that needs our own redirection.  See additional response.

> Is there just one Program Files folder on ARM64 machines❔

On ARM64 machines both x64 and arm64 installers get a single `Program Files` that they need to share.  There is still `Program Files (x86)` for x86 installers, but it is the responsibility of the installer itself to deal with side-by-side on ARM64.